addons/survey/static/src/views/widgets/set_flag_fields/set_flag_fields.js

```diff
@@ -0,0 +1,84 @@
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { booleanField, BooleanField } from "@web/views/fields/boolean/boolean_field";
+import { integerField, IntegerField } from "@web/views/fields/integer/integer_field";
+
+import { useEffect, useRef } from "@odoo/owl";
+
+export class IntegerSetFlag extends IntegerField {
+    static props= {
+        ...IntegerField.props,
+        flagFieldName: { type: String },
+        referenceValue: { type: Number },
+    }
+    /**
+     * @override
+     */
+    setup() {
+        super.setup(...arguments);
+        const inputRef = useRef("numpadDecimal");
+        const onChange = async () => {
+            await this.props.record._update({
+                [this.props.flagFieldName]: parseInt(this.formattedValue) !== this.props.referenceValue}
+            );
+        }
+        useEffect(
+            (inputEl) => {
+                if (inputEl) {
+                    inputEl.addEventListener("change", onChange);
+                    return () => {
+                        inputEl.removeEventListener("change", onChange);
+                    };
+                }
+            },
+            () => [inputRef.el]
+        );
+    }
+}
+
+export const IntegerSetFlagField = {
+    ...integerField,
+    component: IntegerSetFlag,
+    displayName: _t("Integer + set flag"),
```

Comment:
Would update accordingly to other field update (see proposal, if you have a better one feel free :) )